llvm/lib/MC/RepoObjectWriter.cpp

```diff
@@ -930,14 +930,20 @@ uint64_t RepoObjectWriter::writeObject(MCAssembler &Asm,
 
         // The name field of each of the external fixups is pointing into the
         // 'Names' map. Here we turn that into the pstore address of the string.
-        std::for_each(SBegin, SEnd, [](pstore::repo::section_content &Section) {
-          for (auto &XFixup : Section.xfixups) {
-            auto MNC =
-                reinterpret_cast<ModuleNamesContainer::value_type const *>(
-                    XFixup.name.absolute());
-            XFixup.name = MNC->second;
-          }
-        });
+        std::for_each(
+            SBegin, SEnd, [Key](pstore::repo::section_content &Section) {
+              for (auto &XFixup : Section.xfixups) {
+                auto MNC =
+                    reinterpret_cast<ModuleNamesContainer::value_type const *>(
+                        XFixup.name.absolute());
+                XFixup.name = MNC->second;
+                if (XFixup.is_weak)
+                  LLVM_DEBUG(dbgs()
+                             << "fragment " << Key
+                             << " has an xfixup to an external weak symbol '"
+                             << stringViewAsRef(MNC->first) << "' \n");
+              }
```

Comment:
I wouldn’t bind this code into the loop that sets the string addresses. We’ll need to find another technique for setting the string addresses (so that the fields can be less than the full 64 bits) which will mean that this loop will probably disappear.

Reply:
Moved the code to the function `writeSectionData` which write the sections' xfixups.